--- node_modules/@plone/volto/addon-registry.js	2022-03-18 15:02:55.401494934 +0000
+++ node_modules/@plone/volto/addon-registry.js	2022-03-18 15:17:49.027261680 +0000
@@ -105,7 +105,8 @@
       packageJson.name === '@plone/volto'
         ? `${projectRootPath}`
         : `${projectRootPath}/node_modules/@plone/volto`;
-    this.addonNames = (packageJson.addons || []).map((s) => s.split(':')[0]);
+    const addOnNames = process.env.ADDONS ? process.env.ADDONS.split(';') : packageJson.addons;
+    this.addonNames = (addOnNames || []).map((s) => s.split(':')[0]);
     this.packages = {};
     this.customizations = new Map();
 
@@ -114,7 +115,7 @@
     this.initTestingPackages();
 
     this.dependencyGraph = buildDependencyGraph(
-      packageJson.addons || [],
+      addOnNames || [],
       (name) => {
         this.initPublishedPackage(name);
         return this.packages[name].addons || [];
